### ISSUE TITLE
Upgrade from http.ResponseWriter, *http.Request to mangos.Socket

### ISSUE THREAD

**Issue by cameronelliott** [nanomsg/mangos#210]
Forgive me if I missed a way to do this.

I know it's possible to upgrade from a "http.ResponseWriter, *http.Request" pair in Gorilla
to a gorilla.Conn.
I also understand that mangos uses gorilla underneath for it's websockets, right?
It there a way I can upgrade  "http.ResponseWriter, *http.Request" pair to 
a mangos.Socket somehow?

If not, should I consider writing a PR ?

**Comment by gdamore** [comment]
I don’t think so.  The main thing is that we expect websockets as that protocol is defined.  If you can provide the same semantic it might work but both parties in the connection will need to collaborate to make that happen.

Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________
